src/extension.ts

```diff
@@ -164,3 +169,67 @@ async function ensureRuntimeDependencies(extension: vscode.Extension<object>, lo
         return true;
     }
 }
+
+class DocumentLinkProvider implements vscode.DocumentLinkProvider {
+    private _client: LanguageClient;
+
+    constructor(client: LanguageClient) {
+        this._client = client;
+    }
+
+    // Provides the text document with the ranges for document links
+    provideDocumentLinks(document: vscode.TextDocument, token: CancellationToken): vscode.ProviderResult<vscode.DocumentLink[]> {
+        return this._findDocLinks(document);
```

Comment:
Is it worth debouncing this?

Reply:
Not currently sure atm but I can take a look into it

Reply:
I just enjoy musing about things like that. Don't burn much time on investigating; time-box to maybe 30 minutes